cdap-watchdog/src/main/java/co/cask/cdap/logging/save/KafkaLogWriterPlugin.java

```diff
@@ -96,11 +96,13 @@
     Preconditions.checkArgument(retentionDurationDays > 0,
                                 "Log file retention duration is invalid: %s", retentionDurationDays);
 
-    long maxLogFileSizeBytes = cConf.getLong(LoggingConfiguration.LOG_MAX_FILE_SIZE_BYTES, 20 * 1024 * 1024);
+    // Keep max file size a little less than the default HDFS block size (128 MB)
```

Comment:
Even if a file uses space less than the block size, HDFS still allocates space equal to default block size. So it is good to create file sizes a little less than the block size.